pkg/yurtmanager/controller/platformadmin/platformadmin_controller.go

```diff
@@ -440,11 +441,21 @@ func (r *ReconcilePlatformAdmin) reconcileComponent(ctx context.Context, platfor
 			// Refresh the YurtAppSet according to the user-defined configuration
 			yas.Spec.WorkloadTemplate.DeploymentTemplate.Spec = *desiredComponent.Deployment
 
-			if _, ok := yas.Status.PoolReplicas[platformAdmin.Spec.PoolName]; ok {
-				if yas.Status.ReadyReplicas == yas.Status.Replicas {
-					readyDeployment = true
-					if readyDeployment && readyService {
-						readyComponent++
+			for _, summary := range yas.Status.WorkloadSummaries {
+				poolName := ""
+				for i, str := range strings.Split(summary.WorkloadName, "-") {
+					if i == len(strings.Split(summary.WorkloadName, "-"))-1 {
+						break
+					}
+					poolName += str
+					poolName += "-"
+				}
+				if poolName == platformAdmin.Spec.PoolName {
+					if yas.Status.ReadyReplicas == yas.Status.Replicas {
```

Comment:
I think we should use `workloadSummary.AvailableCondition`  to verify component ready or not.

Reply:
I'm just following the original implementation, and I'm not sure if my unauthorized changes here will affect iot sig's usage.